test/OptionalObjectiveTest.js

```diff
@@ -0,0 +1,165 @@
+/*global require*/
+/*global describe*/
+/*global it*/
+
+var assert = require("assert");
+var solver = require("../src/main.js");
+
+describe("Testing if optional objectives are taken into account", function () {
+
+	it("should be able to construct and solve UI model", function () {
+
+		// Elements composing the UI
+		var nUIElements = 0;
+		function UIElement(model){
+			this.varULCornerX = model.addVariable(0, "varULCornerX"+nUIElements, false, false);
+			this.varULCornerY = model.addVariable(0, "varULCornerY"+nUIElements, false, false);
+
+			this.varWidth = model.addVariable(0, "varWidth"+nUIElements, false, false);
+			this.varHeight = model.addVariable(0, "varHeight"+nUIElements, false, false);
+
+			nUIElements += 1;
+		}
+
+
+		function setMinWidth(model, element, minWidth){
+			return [model.greaterThan(minWidth).addTerm(1, element.varWidth)];
+		}
+		function setMaxWidth(model, element, maxWidth){
+			return [model.smallerThan(maxWidth).addTerm(1, element.varWidth)];
+		}
+		function setMinHeight(model, element, minHeight){
+			return [model.greaterThan(minHeight).addTerm(1, element.varHeight)];
+		}
+		function setMaxHeight(model, element, maxHeight){
+			return [model.smallerThan(maxHeight).addTerm(1, element.varHeight)];
+		}
+
+		function setMinXPosition(model, element, minXPosition){
+			return [model.greaterThan(minXPosition).addTerm(1, element.varULCornerX)];
+		}
+		function setMaxXPosition(model, element, maxXPosition){
+			return [model.smallerThan(maxXPosition).addTerm(1, element.varULCornerX)];
+		}
+		function setMinYPosition(model, element, minYPosition){
+			return [model.greaterThan(minYPosition).addTerm(1, element.varULCornerY)];
+		}
+		function setMaxYPosition(model, element, maxYPosition){
+			return [model.smallerThan(maxYPosition).addTerm(1, element.varULCornerY)];
+		}
+
+		function noOverlap(model, element1, element2){
+			var constraintSet = [];
+
+			var xC1 = element1.varULCornerX;
+			var yC1 = element1.varULCornerY;
+			var xC2 = element2.varULCornerX;
+			var yC2 = element2.varULCornerY;
+
+			var W1 = element1.varWidth;
+			var H1 = element1.varHeight;
+			var W2 = element2.varWidth;
+			var H2 = element2.varHeight;
+
+			var MWidth = 1920;
+			var MHeight = 1080;
+
+			var binXC1 = model.addVariable(0, "bx"+xC1.id, true, false);
```

Comment:
Here I would name ```binXC1``` after its function, not after its nature. We want a name that describes its purpose.